scripts/lint_jjb.py

```diff
@@ -155,6 +168,134 @@ def check_retention(job, in_file):
     return return_value
 
 
+# Ensure all jobs have a valid timed triggers entry
+# Structure:
+#   job:
+#     triggers:
+#      - timed: "@daily"
+def check_timed_trigger_value(trigger):
+
+    # If the trigger provided is a NoneType,
+    # then the job has no value, so pass it.
+    if type(trigger) == 'NoneType':
+        return_value = 1
+    # If the trigger provided is not a string,
+    # then it is not valid.
+    if type(trigger) != 'str':
+        return_value = 0
+    # If the CRON macro is used, we do not need
+    # to validate it.
+    elif trigger == "{CRON}":
+        return_value = 1
+    else:
+        # Jenkins supports the use of hashes, denoted by the symbol H,
+        # in the trigger to allow the load to be spread on the system.
+        # The croniter library does not know how to validate them, so
+        # we replace them.
+        # Valid inputs and what they should output after replacement:
+        # 1. "H * * * *" > "* * * * *"
+        # 2. "H(0-29)/10 * * * *" > "0-29/10 * * * *"
+        # 3. "H/10 H(0-11) * H *" > "*/10 0-11 * * *"
+
+        # handle the H(..) pattern first
+        _trigger = re.sub(r'H\((\d-\d+)\)', r'\1', trigger)
+
+        # then handle the H pattern
+        _trigger = re.sub(r'H', r'*', _trigger)
+
+        if croniter.is_valid(_trigger):
+            return_value = 1
+        else:
+            return_value = 0
+
+    return return_value
+
+
+# Ensure all jobs have a valid timed triggers entry
+# Structure:
+#   job:
+#     triggers:
+#      - timed: "@daily"
+#      - timed: "{CRON}"
+def check_timed_trigger_list(data, in_file):
+    error_message = ("{f}/{n}: Valid timer trigger not found."
+                     " triggers: {t}\n"
+                     .format(n=data['name'], f=in_file,
+                             t=data.get('triggers')))
+
+    timed_trigger_list = jmespath.search('triggers[*].timed', data)
+
+    return_value = 0
+    for trigger in timed_trigger_list or []:
+        if check_timed_trigger_value(trigger):
+            sys.stderr.write(error_message)
```

Comment:
Can you have the error message print the actual offending trigger?  That way we're not just spitting out the same error message each time there's an offender found.

Reply:
Done.